tests/integration/test_mongodb.py

```diff
@@ -39,10 +39,14 @@
 pytestmark = pytest.mark.asyncio()
 
 
-class ExampleDto(BaseModelWithId):
-    """Example DTO with an auto-generated ID."""
+class ExampleDto(BaseModel):
+    """Example DTO with an auto-generated ID.
+
+    Does not use `UUID4` type in order to facilitate testing.
+    """
 
     model_config = ConfigDict(frozen=True)
+    id: str = UUID4Field(description="The ID of the resource.")
```

Comment:
The tests should reflect the real/recommended usage of the provided functionality, both in order to test real world usage and since they serve as a "living documentation" on how to use the functionality.

Therefore we should really run the tests with a `UUID4` type here. Unless we recommend that users should also use the `str` type. Which I think they shouldn't, since the UUID4 validation provides additional safety that invalid values can never make their way into the database. (This was actually one of the attack vectors in the penetration tests!)

Reply:
Good point. The tests have been updated to use UUID4